hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -206,6 +216,30 @@ public MapConfig setInMemoryFormat(InMemoryFormat inMemoryFormat) {
         return this;
     }
 
+    /**
+     * Gets the {@link EvictionConfig} instance of the eviction
+     * configuration for this {@link IMap}.
+     *
+     * @return the {@link EvictionConfig}
+     * instance of the eviction configuration
+     */
+    public EvictionConfig getEvictionConfig() {
+        return evictionConfig;
+    }
+
+    /**
+     * Sets the {@link EvictionConfig} instance for eviction
+     * configuration for this {@link IMap}.
+     *
+     * @param evictionConfig the {@link EvictionConfig}
+     *                       instance to set for the eviction configuration
+     * @return current map config instance
+     */
+    public MapConfig setEvictionConfig(EvictionConfig evictionConfig) {
```

Comment:
Users will now be able to configure a custom eviction policy for `IMap` by:
 - setting a custom `MapEvictionPolicy`
 - setting an `EvictionConfig` with a custom `Comparator`

Even without considering custom user supplied eviction policies, there are out-of-the-box `MapEvictionPolicy` implementations which overlap with `EvictionPolicy` configurations (eg `com.hazelcast.map.eviction.LFUEvictionPolicy` and `com.hazelcast.config.EvictionPolicy.LFU`). For example how will a map configured with following snippet behave?

```
config.getMapConfig("te")
                .setMapEvictionPolicy(new LRUEvictionPolicy())
                .getEvictionConfig().setEvictionPolicy(EvictionPolicy.RANDOM)...
```

Can `MapEvictionPolicy` be removed in favor of using `EvictionConfig` only, so there is a single point of configuration and avoid confusion?

Reply:
yes it should be single point of configuration, to make it more explicit, i was planning to send it later, when this one got merged.